hyperapp.d.ts

```diff
@@ -0,0 +1,40 @@
+export as namespace hyperapp
+
+// VirtualNode part
+
+export interface VirtualNode<Data> {
+  tag: string
+  data?: Data
+  children: VirtualElement<{}>[]
+}
+
+export type VirtualElement<Data> = VirtualNode<Data> | string
+
+export interface Component<Data> {
+  (data?: Data, ...children: VirtualElement<{}>[]): VirtualNode<Data>
+}
+
+export function h<Data>(
+  tag: Component<Data> | string,
+  data?: Data,
+  children?: VirtualElement<{}>[] | VirtualElement<{}> | number
```

Comment:
I dont know how this is a problem in fact. Performance ? 

You are saying `h('div', null, 'Hi.')` is less performant than `h('div', null, ['Hi.'])` ?
And what if I still want to use this feature like way, I will get typing/editor error/warning even if the function allow the use of it ? 
I think this is very effortless to types it. And let users to choose you prefered way to do as both exist. 

Let's write a performance warning into the doc and refer it in the JSdoc.
```jsx
/** Description...
 * @see {@link http://github.com/hyperapp/hyperapp/... | Performance issue}
 */
h(....
```